src/mixins/collection.mixin.ts

```diff
@@ -81,87 +72,86 @@
      *                   when no `context` argument is given
      *
      * @param {Object} context Context (aka thisObject)
-     * @return {Self} thisArg
-     * @chainable
      */
-    forEachObject: function (callback, context) {
-      var objects = this.getObjects();
-      for (var i = 0; i < objects.length; i++) {
-        callback.call(context, objects[i], i, objects);
-      }
-      return this;
-    },
+    forEachObject<T>(
+      callback: (
+        this: T | undefined,
+        object: FabricObject,
+        index: number,
+        array: FabricObject[]
+      ) => any,
+      context?: T
+    ) {
+      this.getObjects().forEach(callback.bind(context));
```

Comment:
note for andrea, try to remove the context and the bind.
Does it make sense with fat arrows?
We either don't use it, or use with prebound functions.

Reply:
I think we can drop context

Reply:
I think I tried it and tested but decided not to commit

Reply:
removed now